File: ts/Customer.ts
export default class Customer {}

export class Dog extends Customer {

  public run(meters: number) {
    console.log("Running " + meters + " meters");
  }

  public swim(meters: number) {
    console.log("Swimming " + meters + " meters");
  }

}

export class Human extends Customer {

  public run(meters: number) {
    console.log("Running " + meters + " meters");
  }

  public lift(kg: number) {
    console.log("Lifting " + kg + " kg");
  }

}

File: java/Customer.java
public class Customer {

  public static class Dog extends Customer {

    public void run(int meters) {
      System.out.println("Running " + meters + " meters");
    }

    public void swim(int kg) {
      System.out.println("Swimming " + kg + " meters");
    }

  }

  public static class Human extends Customer {

    public void run(int meters) {
      System.out.println("Running " + meters + " meters");
    }

    public void lift(int kg) {
      System.out.println("Lifting " + kg + " kg");
    }

  }

}